src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -438,9 +438,9 @@ private boolean isFilteredExactSearchPreferred(final int filterIdsCount) {
          * TODO we can have a different MAX_DISTANCE_COMPUTATIONS for binary index as computation cost for binary index
          * is cheaper than computation cost for non binary vector
          */
-        return KNNConstants.MAX_DISTANCE_COMPUTATIONS >= filterIdsCount * (knnQuery.getVectorDataType() == VectorDataType.FLOAT
-            ? knnQuery.getQueryVector().length
-            : knnQuery.getByteQueryVector().length);
+        return KNNConstants.MAX_DISTANCE_COMPUTATIONS >= filterIdsCount * (knnQuery.getVectorDataType() == VectorDataType.BINARY
+            ? knnQuery.getByteQueryVector().length
+            : knnQuery.getQueryVector().length);
```

Comment:
can we split into individual VectorDataType check to avoid else part always assumes it is non byte?

Reply:
updated